File: cognipy/CogniPyLib/CogniPyStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CogniPy
{
    public enum StatementType { Concept, Rule, Role, Instance, Annotation, Constraint }

    public class CogniPyStatement
    {
        public string CnlStatement { get; set; }
        
        public HashSet<string> Concepts { get; set; }
        
        public HashSet<string> Instances { get; set; }
        
        public HashSet<string> Roles { get; set; }
        
        public HashSet<string> DataRoles { get; set; }
        
        public StatementType Type { get; set; }
    }
}

File: cognipy/cognipylib/cognipy/CogniPyStatement.java
package cognipy;

import java.util.*;

public class CogniPyStatement
{
	private String CnlStatement;
	public final String getCnlStatement()
	{
		return CnlStatement;
	}
	public final void setCnlStatement(String value)
	{
		CnlStatement = value;
	}

	private HashSet<String> Concepts;
	public final HashSet<String> getConcepts()
	{
		return Concepts;
	}
	public final void setConcepts(HashSet<String> value)
	{
		Concepts = value;
	}

	private HashSet<String> Instances;
	public final HashSet<String> getInstances()
	{
		return Instances;
	}
	public final void setInstances(HashSet<String> value)
	{
		Instances = value;
	}

	private HashSet<String> Roles;
	public final HashSet<String> getRoles()
	{
		return Roles;
	}
	public final void setRoles(HashSet<String> value)
	{
		Roles = value;
	}

	private HashSet<String> DataRoles;
	public final HashSet<String> getDataRoles()
	{
		return DataRoles;
	}
	public final void setDataRoles(HashSet<String> value)
	{
		DataRoles = value;
	}

	private StatementType Type = StatementType.values()[0];
	public final StatementType getType()
	{
		return Type;
	}
	public final void setType(StatementType value)
	{
		Type = value;
	}
}